pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -280,26 +277,34 @@ func NewNodeLifecycleController(
 	useTaintBasedEvictions bool,
 	taintNodeByCondition bool) (*Controller, error) {
 
-	if kubeClient == nil {
-		klog.Fatalf("kubeClient is nil when starting Controller")
+	if len(tenantPartitionManagers) == 0 {
```

Comment:
in the case of scale-up, could len(tenantPartitionManagers) == 0? or tenantPartitionManagers also covers the scale-up case?

Reply:
In Scale-Up the len(tenantPartitionManagers) will be 1. 

Because of the following code change in func startNodeLifecycleController() in cmd/kube-controller-manager/app/core.go, the tenantPartitionManagers will never be empty.:

if len(ctx.ComponentConfig.NodeLifecycleController.TenantServers) > 0 {
		tpAccessors, err = nodeutil.GetTenantPartitionManagersFromServerNames(ctx.ComponentConfig.NodeLifecycleController.TenantServers, ctx.Stop)
	} else {
		tpAccessors, err = nodeutil.GetTenantPartitionManagersFromKubeClients([]clientset.Interface{kubeclient}, ctx.Stop)
	}